document-portal/document-portal-fuse.c

```diff
@@ -1897,7 +1899,26 @@ xdp_fuse_open (fuse_req_t req,
     return;
 
   path = fd_to_path (inode->physical->fd);
-  fd = open (path, open_flags, 0);
+  if (open_flags & O_NOFOLLOW)
+    {
+      ssize_t res;
+      char actual_path[PATH_MAX];
+      /*
+       * `path` is a path to the fd entry in `/proc`, which is a symlink
+       * to the actual file. Opening it directly with `O_NOFOLLOW` will
+       * fail. So we should resolve it first then we can honour the no
+       * follow flag.
+       */
+      res = readlink (path, actual_path, sizeof (actual_path));
+      if (res == sizeof (actual_path))
+        return xdp_reply_err (op, req, ENAMETOOLONG);
+      if (res < 0)
+        return xdp_reply_err (op, req, errno);
+
+      fd = open (actual_path, open_flags, 0);
+    }
+  else
+    fd = open (path, open_flags, 0);
```

Comment:
Okay, I think code-wise this is fine now. I'll propose the following style change though:

```c
  path = fd_to_path (inode->physical->fd);

  /* `path` is a path to the fd entry in `/proc`, which is a symlink
   * to the actual file. Opening it directly with `O_NOFOLLOW` will
   * fail. So we should resolve it first then we can honour the no
   * follow flag. */
  if (open_flags & O_NOFOLLOW)
    {
      char resolved_path[PATH_MAX] = { 0, };
      ssize_t res;

      res = readlink (path, resolved_path, sizeof (resolved_path));
      if (res == sizeof (resolved_path))
        return xdp_reply_err (op, req, ENAMETOOLONG);
      if (res < 0)
        return xdp_reply_err (op, req, errno);

      g_clear_pointer (&path, g_free);
      path = g_strdup (resolved_path);
    }

  fd = open (path, open_flags, 0);
```

Reply:
I was trying to avoid an extra allocation.

Reply:
done